hs_core/hydroshare/utils.py

```diff
@@ -635,14 +635,25 @@ def convert_file_size_to_unit(size, unit):
         return tbsize
 
 
-def validate_user_quota(user, size):
+def validate_user_quota(user_or_username, size):
     """
     validate to make sure the user is not over quota with the newly added size
-    :param user: the user to be validated
+    :param user_or_username: the user to be validated
     :param size: the newly added file size to add on top of the user's used quota to be validated.
                  size input parameter should be in byte unit
     :return: raise exception for the over quota case
     """
+    if user_or_username:
+        if isinstance(user_or_username, User):
+            user = user_or_username
+        else:
+            try:
+                user = User.objects.get(username=user_or_username)
+            except User.DoesNotExist:
+                user = None
+    else:
+        user = None
```

Comment:
Are you sure we should silently exit when the username doesn't exist?

Reply:
I think it is OK since this function is for validating user quota to warn the user for over-quota cases, so if pass in user does not exist, there is nothing to be validated against. We could add a log here but I don't think it is necessary in this specific case.